src/main.tsx

```diff
@@ -1,60 +1,103 @@
 import {
   resetNotification,
   selectNotification,
 } from "./features/notifications/notification";
+import Unauthorised from "./features/unauthorised/view";
 
 import { Box } from "@chakra-ui/react";
 
 import SimpleSidebar from "./components/sideBar/sideBar";
 
+import { ReactElement, useEffect } from "react";
+import { useAppDispatch, useAppSelector } from "./app/hooks";
 import { ProblemDetailsModal } from "./componentsCommon/problemDetailsModal/problemDetailsModal";
+import { DetectLoggedInUser } from "./features/authentication/authenticationSlice";
 import JsonPlaceHolderView from "./features/jsonPlaceholderAPI/JsonPlaceHolder";
-import { useAppDispatch, useAppSelector } from "./app/hooks";
-import PlaygroundView from "./features/playground/playgroundView";
-import ConfigTableExampleView from "./features/configTable/configTableExample";
-import MovieTableView from "./features/paginatedTable/moviePage";
-import VideoGamesTableView from "./features/formExample/videoGamesPage";
+import { AuthenticationCallback } from "./lib/authentication/authenticationCallback";
+import { useAuthContext } from "./lib/authentication/authenticationContext";
 import { mainSections } from "./siteMap/mainSections";
-import { useEffect } from "react";
+
+//export const authProvider = new Auth0AuthProvider(authConfig);
+
+function initPath(
+  path: string,
+  element: () => React.ReactNode,
+  authenticatedOnly: boolean = true,
+  userIsLogged: boolean = true
+): ReactElement {
+  const routeElement = element();
+  const finalElement =
+    (authenticatedOnly && userIsLogged) || !authenticatedOnly ? (
+      routeElement
+    ) : (
+      <Unauthorised />
+    );
+  return <Route path={path} element={finalElement} />;
+}
 
 const Component = () => {
+  const { context, isLogged } = useAuthContext();
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    dispatch(DetectLoggedInUser());
+  }, [dispatch]);
+
+  const routes = [] as React.ReactNode[];
+  mainSections.forEach((x) =>
+    x.subsections.forEach((s) => {
+      if (s.subsections && s.subsections.length > 0) {
+        s.subsections.forEach((sub) => {
+          if (sub.component && sub.path) {
+            routes.push(
+              initPath(
+                x.path + s.path + sub.path,
+                sub.component,
+                sub.authenticatedOnly,
+                isLogged
+              )
+            );
+          }
+        });
+      } else if (s.component && s.path) {
+        routes.push(
+          initPath(x.path + s.path, s.component, s.authenticatedOnly, isLogged)
+        );
+      }
+    })
+  );
+
   return (
     <>
       <Routes>
-        <Route index path="/" element={<Navigate to="/test" />} />
+        <Route
+          index
+          path="/"
+          element={
+            <AuthenticationCallback
+              entryPoint={<JsonPlaceHolderView />} //Change this to your preffered entry point
```

Comment:
Entry point should be either defined in SiteMap and should be rendered only once. I assume the component JsonPlaceHolderView is also placed in the Sitemap as can have all the settings like AuthorizedOnly or Permissions.

Since routes can be nested, I'm sure this can be made hassle free by rendering {routes} as subroutes of via 'layout' routes that don't add to the path.

see https://reactrouter.com/en/main/route/route#layout-routes or https://reactrouter.com/en/main/route/route#index 

see